doc/neps/nep-0042-new-dtypes.rst

```diff
@@ -0,0 +1,1149 @@
+avoid the potential for conflicts and .
+It is the DType implementors responsibility to be careful about this and use
+the flag to disable registration when in doubt.
+
+**Alternatives:**
+
+The above proposes to add a global mapping, however, initially limiting it
+to types deriving from a NumPy subclass (and a fixed set of Python types).
+This could be relaxed in the future.
+Alternatively, we could rely on the scalar belonging to the user dtype to
+implement ``scalar.dtype`` or similar.
+Initially, at the moment the exact implementation shall be *undefined*.
+Scalars will have to derive from a NumPy scalar, and provide at least a
+``.dtype`` attribute. Although, this may not be used within NumPy.
+Any such scalar shall be expected to have a corresponding DType.
+
+An initial alternative suggestion was to use a two-pass approach instead.
+The first pass would only find the correct DType class, and the second pass
+would then find correct dtype instance (the second pass is often not necessary).
+The advantage of this is that the DType class information is vital for universal
+functions to decide which loop to execute.
+The first pass would provide the full information necessary for value based
+casting currently implemented for scalars, giving even the possibility of
+expanding it to e.g. list inputs ``np.add(np.array([8], dtype="uint8"), [4])``
+giving a ``uint8`` result.
+This is mainly related to the question to how the common dtype is found above.
+It seems unlikely that this is useful, and similar to a global, could be
+added later if deemed necessary.
+
+**Further Issues or Discussion:**
+
+While it is possible to create e.g. a DTypes such as Categorical, array,
+or vector which can only be used if `dtype=DType` is provided, if this
+is necessary these will not roundtrip correctly when converted back
+and forth::
+
+    np.array(np.array(1, dtype=Categorical)[()])
+
+requires to pass the original ``dtype=Categorical`` or returns an array
+with dtype ``object``.
+While a general limitation, the roundtripping shall always be possible if
+``dtype=old_dtype`` is provided.
+
+**Example:**
+
+The current datetime DType requires a ``__discover_descr_from_pyobject__``
+which returns the correct unit for string inputs.  This allows it to support
+the current::
+
+    np.array(["2020-01-02", "2020-01-02 11:24"], dtype="M8")
+
+By inspecting the date strings. Together with the below common dtype
+operation this allows it to automatically find that the datetime64 unit
+should be "minutes".
+
+
+Common DType Operations
+^^^^^^^^^^^^^^^^^^^^^^^
+
+Numpy currently provides functionality of ``np.result_type`` and
+``np.promote_types`` (while more common, ``np.result_type`` has some more
+complex logic due to implementing value based promotion [value_based]_).
+(Note that the name ``np.common_type`` is associated with the scalars and not
+directly with dtypes and is of limited usefulness)
```

Comment:
```suggestion
Numpy currently provides functions like ``np.result_type`` and
``np.promote_types`` for determining common types [#ct1]_ [#ct2]_

.. [#ct1] While ``np.promote_types` is more common, ``np.result_type`` has some more complex logic due to implementing value-based promotion [value_based]_.
.. [#ct2] Note that the name ``np.common_type`` is associated with the scalars and not directly with dtypes and is of limited usefulness)
```
Since you have two parenthetical statements here, you could try turning them into footnotes instead (I think the syntax I suggested is at least close to how you can do that in rST)

Reply:
Just deleted half of it :).